server/graphql/v2/mutation/AgreementMutations.ts

```diff
@@ -0,0 +1,149 @@
+
+import twoFactorAuthLib from '../../../lib/two-factor-authentication';
+import { UploadedFile } from '../../../models';
+import AgreementModel from '../../../models/Agreement';
+import { checkRemoteUserCanUseHost } from '../../common/scope-check';
+import { Unauthorized } from '../../errors';
+import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
+import { AgreementReferenceInput, fetchAgreementWithReference } from '../input/AgreementReferenceInput';
+import { Agreement } from '../object/Agreement';
+
+// TODO(ESM): Move this to standard ESM imports once available and remove the promise
+export default new Promise(resolve => {
+  import('graphql-upload/GraphQLUpload.mjs').then(({ default: GraphQLUpload }) => {
+    resolve({
+      addAgreement: {
+        type: new GraphQLNonNull(Agreement),
+        description: 'Add an agreement for the given host account. Scope: "host".',
+        args: {
+          title: {
+            type: new GraphQLNonNull(GraphQLNonEmptyString),
+            description: 'Agreement title.',
+          },
+          expiresAt: {
+            type: GraphQLDateTime,
+            description: 'Optional date in which this agreement expires.',
+          },
+          host: {
+            type: new GraphQLNonNull(AccountReferenceInput),
+            description: 'Host where the agreement will be created.',
+          },
+          account: {
+            type: new GraphQLNonNull(AccountReferenceInput),
+            description: 'Account that is a party in this agreement',
+          },
+          attachment: {
+            type: GraphQLUpload,
+            description: 'Agreement attachment',
+          },
+        },
+        async resolve(_: void, args, req: express.Request): Promise<AgreementModel> {
+          checkRemoteUserCanUseHost(req);
+
+          const host = await fetchAccountWithReference(args.host, {
+            throwIfMissing: true,
+          });
+
+          if (!req.remoteUser.isAdminOfCollective(host)) {
+            throw new Unauthorized('Only host admins can create agreements');
+          }
+
+          await twoFactorAuthLib.enforceForAccount(req, host);
+
+          const account = await fetchAccountWithReference(args.account, {
+            throwIfMissing: true,
+          });
+
+          const attachment: Promise<FileUpload> = args.attachment;
+
+          let uploadedFile;
+          if (attachment) {
+            uploadedFile = await UploadedFile.uploadGraphQl(await attachment, 'AGREEMENT_ATTACHMENT', req.remoteUser);
+          }
+
+          const agreement = await AgreementModel.create({
+            title: args.title,
+            expiresAt: args.expiresAt,
+            HostCollectiveId: host.id,
+            CollectiveId: account.id,
+            UserId: req.remoteUser.id,
+            UploadedFileId: uploadedFile?.id,
+          });
+
+          return agreement;
+        },
+      },
+      editAgreement: {
```

Comment:
Should we allow attachment updates here? E.g adding an attachment to the agreement, removing one that is there, or uploading/changing to a new file